[Flaky test] [cmd/opampsupervisor] TestSupervisorRemoteConfigApplyStatus flaky

### Component(s)

cmd/opampsupervisor

### Describe the issue you're reporting

https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/11746011659/job/32725597882

```
=== RUN   TestSupervisorRemoteConfigApplyStatus
    e2e_test.go:163: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/cmd/opampsupervisor/e2e_test.go:163
        	            				D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/cmd/opampsupervisor/e2e_test.go:1475
        	Error:      	Received unexpected error:
        	            	yaml: line 15: did not find expected hexdecimal number
        	Test:       	TestSupervisorRemoteConfigApplyStatus
    e2e_test.go:132: Shutting down
--- FAIL: TestSupervisorRemoteConfigApplyStatus (5.01s)
```

--- a/cmd/opampsupervisor/e2e_test.go
+++ b/cmd/opampsupervisor/e2e_test.go
@@ -188,7 +188,7 @@ func getSupervisorConfig(t *testing.T, configType string, extraConfigData map[st
 		"goos":        runtime.GOOS,
 		"goarch":      runtime.GOARCH,
 		"extension":   extension,
-		"storage_dir": t.TempDir(),
+		"storage_dir": strings.ReplaceAll(t.TempDir(), "\\", "\\\\"),
 	}
 
 	for key, val := range extraConfigData {

